core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,729 @@ The document defines storage capability requirements in the following areas:
+12. Attempt to delete a cryptographic key from the HSM.
+13. Verify that the BPI communicates with the HSM to delete the key securely.
+14. Attempt to perform a cryptographic processing operation available in the chosen test HSM using a cryptographic key created during the test in the HSM.
+15. Verify that the cryptographic processing operation produced a cryptographic valid result.
+
+Test Passing Criteria: The test will pass if
+
+* The BPI redirects the BPI Subject to a login page or displays a message indicating that authentication is required.
+* Successful login grants access to the BPI.
+* The BPI can generate a cryptographic key using the HSM.
+* The BPI can store a cryptographic key securely in the HSM.
+* The BPI can retrieve a cryptographic key securely from the HSM.
+* The BPI can delete a cryptographic key securely from the HSM.
+* The BPI can perform cryptographic processing operations available in the HSM.
+
 ## 7.2. BPI Storage Privacy
+###### BPISTORAGEPRIVACY
 
 As has been done throughout this document, there are BPI layer-specific privacy requirements for layers and/or components which are listed below. It is assumed that the BPI Storage capabilities are required to fulfill all BPI privacy requirements in section [2 Design and Architecture](#2-design-and-architecture).
 
 #### **[R288]**	
 Personal Identifiable Information (PII) MUST NOT be stored in a BPI.
 
+[[R288]](#r288) Testability: 
+
+Preconditions:
+
+* A BPI test environment is installed and configured properly.
+* The BPI storage component contains identification methods to prevent storing Personal Identifiable Information (PII).
+* The BPI Subject has appropriate permissions to access the BPI.
+
+Test Steps:
+
+1. Create a record that contains PII, such as a name, address, social security number, or phone number.
+2. Attempt to store the record in the BPI using a write operation method, such as POST or PUT.
+3. Verify that the BPI rejects the record and returns an error message indicating that the record contains PII and cannot be stored.
+4. Create a record that does not contain PII, such as a random string or number.
+5. Attempt to store the record in the BPI using a write operation method, such as POST or PUT.
+6. Verify that the BPI accepts the record and stores it successfully.
+
+Example Methods to identify PII:
+
+* Use regular expressions or data pattern matching to identify common PII data elements in a record.
+* Compare record data with a pre-defined list of PII data elements to identify PII.
+
+Test Passing Criteria: The test will pass if
+
+* The BPI rejects the storage of records that contain Personal Identifiable Information (PII) and returns an error message indicating that PII cannot be stored.
+* The BPI accepts the storage of records that do not contain PII and stores them successfully.
+* The BPI is able to identify PII using the specified methods.
+* The BPI correctly identifies records that contain PII and rejects their storage.
```

Comment:
I think it's reasonable for the BPI to provide a workflow that effectively requires users to enforce this. Identifying PII as a general problem is a very complex task...